cpp/src/arrow/flight/server.cc

```diff
@@ -352,7 +352,7 @@ class DoExchangeMessageWriter : public FlightMessageWriter {
   }
 
   grpc::ServerReaderWriter<pb::FlightData, pb::FlightData>* stream_;
-  ipc::IpcOptions ipc_options_;
+  ::arrow::ipc::IpcOptions ipc_options_;
```

Comment:
This needs to also be IpcWriteOptions. Try building with `-DARROW_NO_DEPRECATED_API=ON`

Reply:
This change seems to break the build on s390x, as shown in https://travis-ci.org/github/apache/arrow/jobs/691574657

Reply:
on s390x with `-DARROW_NOW_DEPRECATED_API=ON` and w/o this PR cause the following build error
```
[ 60%] Building CXX object src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/server.cc.o
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc:355:8: error: 'IpcOptions' in namespace 'arrow::ipc' does not name a type
   ipc::IpcOptions ipc_options_;
        ^~~~~~~~~~
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc: In constructor 'arrow::flight::{anonymous}::DoExchangeMessageWriter::DoExchangeMessageWriter(grpc::ServerReaderWriter<arrow::flight::protocol::FlightData, arrow::flight::protocol::FlightData>*)':
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc:281:26: error: class 'arrow::flight::{anonymous}::DoExchangeMessageWriter' does not have any field named 'ipc_options_'
       : stream_(stream), ipc_options_(ipc::IpcOptions::Defaults()) {}
                          ^~~~~~~~~~~~
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc:281:44: error: 'arrow::ipc::IpcOptions' has not been declared
       : stream_(stream), ipc_options_(ipc::IpcOptions::Defaults()) {}
                                            ^~~~~~~~~~
In file included from /home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server_auth.h:25:0,
                 from /home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.h:29,
                 from /home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc:21:
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc: In member function 'virtual arrow::Status arrow::flight::{anonymous}::DoExchangeMessageWriter::Begin(const std::shared_ptr<arrow::Schema>&)':
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc:291:18: error: 'ipc_options_' was not declared in this scope
         *schema, ipc_options_, &dictionary_memo_, &schema_payload.ipc_message));
                  ^
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/status.h:57:62: note: in definition of macro 'ARROW_RETURN_NOT_OK'
     ::arrow::Status __s = ::arrow::internal::GenericToStatus(status); \
                                                              ^~~~~~
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc:290:5: note: in expansion of macro 'RETURN_NOT_OK'
     RETURN_NOT_OK(ipc::internal::GetSchemaPayload(
     ^~~~~~~~~~~~~
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc: In member function 'virtual arrow::Status arrow::flight::{anonymous}::DoExchangeMessageWriter::WriteWithMetadata(const arrow::RecordBatch&, std::shared_ptr<arrow::Buffer>)':
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc:314:53: error: 'ipc_options_' was not declared in this scope
         ipc::internal::GetRecordBatchPayload(batch, ipc_options_, &payload.ipc_message));
                                                     ^
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/status.h:57:62: note: in definition of macro 'ARROW_RETURN_NOT_OK'
     ::arrow::Status __s = ::arrow::internal::GenericToStatus(status); \
                                                              ^~~~~~
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc:313:5: note: in expansion of macro 'RETURN_NOT_OK'
     RETURN_NOT_OK(
     ^~~~~~~~~~~~~
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc: In member function 'arrow::Status arrow::flight::{anonymous}::DoExchangeMessageWriter::EnsureDictionariesWritten(const arrow::RecordBatch&)':
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc:348:36: error: 'ipc_options_' was not declared in this scope
           pair.first, pair.second, ipc_options_, &payload.ipc_message));
                                    ^
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/status.h:57:62: note: in definition of macro 'ARROW_RETURN_NOT_OK'
     ::arrow::Status __s = ::arrow::internal::GenericToStatus(status); \
                                                              ^~~~~~
/home/ishizaki/Arrow/arrow-buildfailure/cpp/src/arrow/flight/server.cc:347:7: note: in expansion of macro 'RETURN_NOT_OK'
       RETURN_NOT_OK(ipc::internal::GetDictionaryPayload(
```

Reply:
How about `::arrow::ipc::IpcWriteOptions ipc_options_;`? It can succeed to compile arrow-flight.

Reply:
Yes, my mistake, thank you both for pointing it out. I've updated this PR.